susemanager-utils/susemanager-sls/src/grains/cpuinfo.py

```diff
@@ -114,3 +116,30 @@ def total_num_cpus():
     sysdev = '/sys/devices/system/cpu/'
     return {'total_num_cpus': len([cpud for cpud in (os.path.exists(sysdev) and os.listdir(sysdev) or list())
                                    if re_cpu.match(cpud)])}
+
+
+def cpu_data():
+    """
+    Returns the cpu model, vendor ID and other data that may not be in the cpuinfo
+    """
+    lscpu = _which_bin(['lscpu'])
+    if lscpu is not None:
+        try:
+            log.debug("Trying lscpu to get CPU data")
+            ret = __salt__['cmd.run_all']('{0} -J'.format(lscpu), output_loglevel='quiet')
+            if ret['retcode'] == 0:
+                data = json.loads(ret["stdout"])
+                name_map = {
+                    "Model name": "cpu_model",
+                    "Vendor ID": "cpu_vendor",
+                    "NUMA node(s)": "cpu_numanodes",
+                    "Stepping": "cpu_stepping",
+                    "Core(s) per socket": "cpu_cores",
+                }
+                values = {name_map[entry["field"][:-1]]: entry["data"] for entry in data.get("lscpu") if entry["field"][:-1] in name_map.keys()}
+                log.debug(values)
+                return values
+            else:
+                log.warn("lscpu does not support -J option")
+        except Exception as error:
```

Comment:
It would be nice to be explicit about the possible exceptions. From looking at the code, I think there can only be a `salt.exceptions.CommandExecutionError` (L129), `ValueError` (or the subclass `json.JSONDecodeError` in Python 3.5+) (L131), or a `TypeError` (L139). Logging could in theory also cause an error, but that would also be raised inside the exception handling anyway.

The `TypeError` is rather unlikely, it only happens when `data.get("lscpu")` returns `None` but JSON didn't throw an error when parsing the command output.